### DISCUSSION TITLE
Welcome to dextool Discussions!

### DISCUSSION THREAD

**Discussion by joakim-brannstrom** [joakim-brannstrom/dextool#1244] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Swati90-coder** [comment]
Hello , 
During dextool installation i see below error : 

[  6%] Generating dextool_mylib.o
/home/sw/dextool/vendor/mylib/source/my/actor/actor.d(69,34): **Error**: copy constructor my.actor.mailbox.Reply.this(return ref Reply a) is not callable using argument types (ulong, VariantN!32LU)
/home/sw/dextool/vendor/mylib/source/my/actor/actor.d(69,34):        cannot pass argument this.data.get().replyId of type ulong to parameter return ref Reply a
/home/sw/dextool/vendor/mylib/source/my/actor/utility/limiter.d(57,9): **Error**: template instance my.actor.actor.Promise!(Token) error instantiating
/home/sw/dextool/vendor/mylib/source/my/actor/utility/limiter.d(64,12): **Error**: template instance my.actor.actor.RequestResult!(Token) error instantiating
/home/sw/dextool/vendor/mylib/source/my/profile.d(138,23): Error: template object.update cannot deduce function from argument types !()(Duration[string], string, Duration delegate() pure nothrow @nogc @safe, void delegate(ref Duration a) pure nothrow @nogc @safe), candidates are:
/usr/lib/ldc/x86_64-linux-gnu/include/d/object.d(2719,6):        update(K, V, C, U)(ref V[K] aa, K key, scope C create, scope U update)
  with K = string,
       V = Duration,
       C = Duration delegate() pure nothrow @nogc @safe,
       U = void delegate(ref Duration) pure nothrow @nogc @safe
  must satisfy the following constraint:
       is(typeof(update(aa[K.init])) : V)
/home/sw/dextool/vendor/mylib/source/my/profile.d(143,23): **Error**: template object.update cannot deduce function from argument types !()(Duration[string], string, Duration delegate() pure nothrow @nogc @safe, void delegate(ref Duration a) pure nothrow @nogc @safe), candidates are:
/usr/lib/ldc/x86_64-linux-gnu/include/d/object.d(2719,6):        update(K, V, C, U)(ref V[K] aa, K key, scope C create, scope U update)
  with K = string,
       V = Duration,
       C = Duration delegate() pure nothrow @nogc @safe,
       U = void delegate(ref Duration) pure nothrow @nogc @safe
  must satisfy the following constraint:
       is(typeof(update(aa[K.init])) : V)
make[2]: *** [vendor/CMakeFiles/dextool_mylib.dir/build.make:111: vendor/dextool_mylib.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:826: vendor/CMakeFiles/dextool_mylib.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

on commenting the lines of error , mylib is installed , but issue is seen in dextool-analyze .
The library installed :
cmake
cmake
llvm-10
llvm-10-dev
sqlite3
sqlite-3-dev
clang-10
libclang-10-dev

**Reply by joakim-brannstrom** [reply]
Sorry for the late reply. IRL busy.

The reason for the errors is most likely that the D compiler isn't compatible. You can see what versions are supported in [README](https://github.com/joakim-brannstrom/dextool/blob/master/README.md). I do not spend too much work on keeping dextool working with older versions of the compiler because of lack of resources for that when it is "easy" to the compiler :)
